src/components/pages/ResultQuizPage/ResultQuizPage.tsx

```diff
@@ -4,42 +4,56 @@ import { QuizConfigContainer } from "./QuizConfigContainer/QuizConfigContainer";
 import { ResultNumberField } from "./ResultNumberField/ResultNumberField";
 import { ResultContainer } from "./ResultContainer/ResultContainer";
 import { useNavigate } from "react-router-dom";
-import { MAIN, START } from "../../Router/routes";
+import { MAIN, START} from "../../Router/routes";
 import { useDispatch } from "react-redux";
-import { clearConfiguration } from "../../../redux/slices/configurationSlice";
 import { useSelector } from "react-redux";
-import { clearCorrectAnswers } from "../../../redux/slices/resultSlice";
 import { Store } from "../../../redux/store/interface/store.interface";
-import { clearQuestions } from "../../../redux/slices/questionsSlice";
-import { clearLoadedQuestions } from "../../../redux/slices/loadQuestionsSlice";
+import { clearCorrectAnswers } from "../../../redux/slices/resultSlice";
+import { useClearCurrentQuizData } from "../../../hooks/useClearCurrentQuizData";
+import { useEffect } from "react";
+import { addCurrentQuizResultToStatistics } from "../../../redux/slices/statisticsSlice";
 
 const ResultQuizPage = () => {
     const dispatch = useDispatch();
     const navigate = useNavigate();
-    const result = useSelector((state: Store) => state.result);
+    const correctAnswers = useSelector((state: Store) => state.result.correctAnswers);
     const questions = useSelector((state: Store) => state.questions);
+    const {category, difficulty, type} = useSelector((state: Store) => state.configuration);
+    const clearCurrentQuizData = useClearCurrentQuizData(); 
 
     const onClickRestartQuizHandler = () => {
         dispatch(clearCorrectAnswers());
         navigate(MAIN);
     };
 
     const onClickAnotherQuizHandler = () => {
-        dispatch(clearConfiguration());
-        dispatch(clearCorrectAnswers());
-        dispatch(clearQuestions());
-        dispatch(clearLoadedQuestions())
+        clearCurrentQuizData();
         navigate(START);
     };
 
+    const saveStatistics = () => {
+        dispatch(addCurrentQuizResultToStatistics({
+            questions: questions.length,
+            correctAnswers,
+            category,
+            difficulty, 
+            type
+        }));
+    };
+
+    useEffect(() => {
+        console.log('Mounting result pade ande geting quiz results');
+        saveStatistics()
+    });
```

Comment:
I think this is a bad practice to call useEffect each time
please think about dependency here or about handler to save statistic